core/src/main/java/tachyon/worker/eviction/EvictLFU.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.eviction;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.HashMultimap;
+
+import tachyon.Pair;
+import tachyon.worker.hierarchy.BlockInfo;
+import tachyon.worker.hierarchy.StorageDir;
+
+/**
+ * Used to evict old blocks among several StorageDirs by LFU strategy. LFU (Least Frequently Used)
+ * algorithm evicts blocks with the lowest reference frequency. @see <a
+ * href="http://en.wikipedia.org/wiki/Least_frequently_used"
+ * >http://en.wikipedia.org/wiki/Least_frequently_used</a>
+ */
+public final class EvictLFU extends EvictLFUBase {
+
+  public EvictLFU(boolean lastTier) {
+    super(lastTier);
+  }
+
+  /**
+   * Get StorageDir allocated and also get blocks to be evicted among StorageDir candidates. Since
+   * each StorageTier has only one EvictStrategy but may have more than one eviction event at one
+   * time so it needs to be synchronized.
+   * 
+   * @param storageDirs StorageDir candidates that the space will be allocated in
+   * @param pinList list of pinned file
+   * @param requestBytes requested space size in bytes
+   * @return Pair of StorageDir allocated and blockInfoList which contains information of blocks to
+   *         be evicted, null if no allocated directory is found
+   */
+  @Override
+  public synchronized Pair<StorageDir, List<BlockInfo>> getDirCandidate(StorageDir[] storageDirs,
+      Set<Integer> pinList, long requestBytes) {
+    /** blocks to be evicted */
```

Comment:
/** ... */ is the style of class member comment in javadoc. Here I just use this to distinguish the variable descriptions and general descriptions more clearly.